modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -93,24 +96,32 @@ public int getSize() {
       R result = function.apply(connection);
       connectionSupplier.close();
```

Comment:
We are closing the connectionSupplier here, so no need to save it in a variable and close it later on the close() method.

Reply:
finally